Taken from https://bbs.archlinux.org/viewtopic.php?pid=1796979#p1796979 to
avoid the following inaccurate error message:

Error: SASL(-13): authentication failure: server rejected XOAUTH2:
Warning: SASL reported failure despite successful IMAP authentication. Ignoring...

diff --git a/xoauth2.c b/xoauth2.c
index 9ccca2f..7c3cd69 100644
--- a/xoauth2.c
+++ b/xoauth2.c
@@ -45,6 +45,8 @@ static int xoauth2_client_mech_step(void *conn_context,
     int r = SASL_OK;
 
     if (text->state > 0) {
+        if (serverinlen == 0 || (serverinlen == 1 && serverin[0] == '\0') )
+            return SASL_OK;
         static char buf[1024];
         strncpy(buf, serverin, serverinlen);
         buf[serverinlen < 1023 ? serverinlen : 1023] = '\0';
